### PR TITLE
Quit graphpass on graph with > 1000000 edges in the graph.

### PR DESCRIPTION
**Pull request by greebie**
**The title of this pull-request should be a brief description of what the pull-request fixes/improves/changes. Ideally 50 characters or less.**

#60 

# What does this Pull Request do?

Drops the graphpass job if the edges are greater than 1,000,000.

# How should this be tested?

Unfortunately, this may take some testing to be sure that we are happy with the 1,000,000 edge limit. For example, are there gexfs in our collection that do not potato browsers but yet have 1,000,000 or more edges?

You can use graphpass to get the edge count using the no-save and verbose flags (n & v):
`./graphpass -nv -f {FILENAME}`.  I will do a check myself, but thought I'd push this one in case someone else has time to check a few as well.

# Additional Notes:
This should cover the vast number of problems with files being too large. The only other potential future way we could have huge files is if labels or other data components were super-large.

I do not think we need to resolve this now, but it's worth mentioning for the future.

# Interested parties

@ianmilligan1 @ruebot

### CODE REVIEW

File: src/headers/graphpass.h [hunk 1]
```diff
@@ -40,6 +40,7 @@ char* ug_methods;  /**< METHODS to filter */
 char* ug_OUTPUT;  /**< Folder to output new graphs */
 char* OUTPATH; /**< Path to output folder (DIRECTORY + OUTPUT) */
 igraph_integer_t NODESIZE; /**< Number of Nodes in original graph */
+igraph_integer_t EDGESIZE; /**< Number of Edges in origianl graph */
```
Comment (ianmilligan1) [inline src/headers/graphpass.h]: typo on "origianl"

**Comment by ianmilligan1** [comment]
Sure. We're not in a big hurry, so let me just test this on all our graphml files. I'll record the output.

**Comment by greebie** [comment]
Tested on all the graphs in auk_graphml over 100MB.  So far only 2402 gets cancelled.

**Comment by greebie** [comment]
The latest commit adds the --max-edges (-x) and --max-nodes (-y) flag options.  If there are no flags, it will default to 50k and 1m.

test using: 
`./graphpass -nv -f {FILENAME} --max-edge 90 --max-nodes 60000`

**Comment by ianmilligan1** [comment]
Thanks for the update @greebie. FWIW I'm running this on all our WALK graphmls (~117 of them). I'd like to get a table of nodes/edges so we can make sure to draw the line accordingly.

**Comment by greebie** [comment]
Thanks @ianmilligan1.  I'm good with whatever default values you'd like to work with.

File: src/headers/graphpass.h [hunk 2]
```diff
@@ -75,7 +78,10 @@ igraph_vector_t WEIGHTED; /**< If greater than 0, conducts weighted analysis */
 #define COLOR_BASE "WalkTrapModularity"
 #define PAGERANK_DAMPING 0.85 /**< chance random walk will not restart */
 #define LAYOUT_DEFAULT_CHAR 'f'
-#define MAX_NODES 50000 /**< number of nodes in graph before shut down */
+#define MAX_NODES 50000 /**< default number of nodes in graph before shut down */
+#define MAX_EDGES 1000000 /**< default number of edges in graph before shut down */
```
Comment (ianmilligan1) [inline src/headers/graphpass.h]: As I note in overall comments, recommend changing these `1000000` values to `500000`.

Comment (greebie) [reply src/headers/graphpass.h]: Okay.  I'll lower the default to 500,000.

**Comment by ianmilligan1** [comment]
Can you update the branch, @greebie?

**Pull request merged**